applications/nrf5340_audio/src/modules/sd_card_playback.c

```diff
@@ -0,0 +1,411 @@
+			return ret;
+		}
+	}
+	ret = sd_card_close(&f_seg_read_entry);
+	if (ret < 0) {
+		LOG_ERR("Close file err: %d", ret);
+		return ret;
+	}
+	sd_card_playback_active = false;
+	return 0;
+}
+
+static int sd_card_playback_play_wav(void)
+{
+	int ret;
+	size_t pcm_wav_mono_frame_size = wav_playback_cfg.frame_duration_ms *
+					 wav_playback_cfg.sample_rate / 1000 *
+					 wav_playback_cfg.bit_depth / 8;
+	pcm_frame_size = pcm_wav_mono_frame_size;
+	uint8_t pcm_wav_mono_frame[pcm_wav_mono_frame_size];
+
+	ret = sd_card_open(playback_file_name, &f_seg_read_entry);
+	if (ret < 0) {
+		LOG_ERR("Open SD card file err: %d", ret);
+		return ret;
+	}
+
+	sd_card_playback_active = true;
+	while (sd_card_playback_active) {
+
+		ret = sd_card_read(pcm_wav_mono_frame, &pcm_wav_mono_frame_size, &f_seg_read_entry);
+		if (ret < 0) {
+			LOG_ERR("SD card read err: %d", ret);
+			return ret;
+		}
+		if (pcm_wav_mono_frame_size < pcm_frame_size) {
+			sd_card_playback_active = false;
+		}
+		/* Wait until there is enough space in the ringbuffer */
+		if (sd_card_playback_active) {
+
+			ret = sd_card_playback_load_ringbuf(pcm_wav_mono_frame,
+							    pcm_wav_mono_frame_size);
+			if (ret < 0) {
+				LOG_ERR("Load ringbuf err: %d.", ret);
+				return ret;
+			}
+		}
+	}
+	ret = sd_card_close(&f_seg_read_entry);
+	if (ret < 0) {
+		LOG_ERR("SD card close file err: %d", ret);
+		return ret;
+	}
+	sd_card_playback_active = false;
+	return 0;
+}
+
+static void sd_card_playback_thread(void *arg1, void *arg2, void *arg3)
+{
+	int ret;
+	while (1) {
+		if (!sw_codec_is_initialized()) {
+			k_msleep(200);
+			continue;
+		}
+		k_sem_take(&m_sem_playback, K_FOREVER);
+		switch (playback_file_format) {
+		case SD_CARD_PLAYBACK_WAV:
+			ret = sd_card_playback_play_wav();
+			if (ret < 0) {
+				LOG_ERR("Wav playback err: %d", ret);
+			}
+			break;
+		case SD_CARD_PLAYBACK_LC3:
+			ret = sd_card_playback_play_lc3();
+			if (ret < 0) {
+				LOG_ERR("LC3 playback err: %d", ret);
+			}
+			break;
+		}
+	}
+}
+
+bool sd_card_playback_is_active(void)
+{
+	return sd_card_playback_active;
+}
+
+void sd_card_playback_wav(char *filename, uint32_t frame_duration_ms, uint8_t bit_depth,
+			  enum sd_playback_sample_rates sample_rate)
+{
+	playback_file_format = SD_CARD_PLAYBACK_WAV;
+	playback_file_name = filename;
+	wav_playback_cfg.frame_duration_ms = frame_duration_ms;
+	wav_playback_cfg.bit_depth = bit_depth;
+	wav_playback_cfg.sample_rate = sample_rate;
+	k_sem_give(&m_sem_playback);
+}
+
+int sd_card_playback_lc3(char *filename)
+{
+	if (!sw_codec_is_initialized()) {
+		LOG_ERR("Sw codec not initialized");
+		return -EACCES;
+	}
+	playback_file_format = SD_CARD_PLAYBACK_LC3;
+	playback_file_name = filename;
+	k_sem_give(&m_sem_playback);
+	return 0;
+}
+
+int sd_card_playback_mix_with_stream(void *const pcm_a, size_t pcm_a_size)
+{
+	int ret;
+	uint8_t pcm_b[pcm_frame_size];
+	size_t write_size = pcm_frame_size;
+
+	ret = sd_card_playback_buffer_set(pcm_b, &write_size);
```

Comment:
```suggestion
	ret = sd_card_playback_buffer_set(pcm_b, &read_size);
```
... or am I just mixing up the read/write functions now?